compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/contracts/FirAbstractContractResolveTransformerDispatcher.kt

```diff
@@ -61,12 +62,25 @@ abstract class FirAbstractContractResolveTransformerDispatcher(
     private val regularDeclarationsTransformer = FirDeclarationsResolveTransformer(this)
 
     private var contractMode = true
+    private var insideContractDescription = false
 
     override fun transformAnnotation(annotation: FirAnnotation, data: ResolutionMode): FirStatement {
         return annotation
     }
 
     override fun transformAnnotationCall(annotationCall: FirAnnotationCall, data: ResolutionMode): FirStatement {
+        // Annotation calls inside contracts are not allowed and at this point we are also
+        // unable to resolve them. We indicate this by returning a resolved error type.
+        if (insideContractDescription) {
+            val typeDiagnostic = ConeSimpleDiagnostic(
+                "Cannot infer annotation call type during CONTRACTS phase.",
+                DiagnosticKind.InferenceError
+            )
+            val errorType = ConeErrorType(typeDiagnostic)
+
+            annotationCall.replaceAnnotationTypeRef(errorType.toFirResolvedTypeRef(annotationCall.annotationTypeRef.source))
+        }
```

Comment:
Here I was also experimenting with returning a `FirErrorAnnotationCall` with a custom error message that says `"Annotations calls are prohibited inside contract descriptions"`, though the message ended up being a duplicate of what is reported by `FirContractChecker`.

Also, since `FirErrorAnnotationCall` still needs a resolved type to not crash, we would still see the current error, so I ended up sticking to creating the error type only without changing the node too.

Is any additional change desired here?

Reply:
Your approach is correct

Message in `ConeSimpleDiagnostic` is mostly used for debug purposes

Reply:
I meant an error message that we put into `FirErrorsDefaultMessages.kt`.

Right now the compiler output is the following:
```
e: Error in contract description: illegal element in contract description
e: Inference error.
e: Syntax error: Expecting an expression.
```

With my message it was
```
e: Error in contract description: illegal element in contract description
e: Annotation calls are prohibited inside contract descriptions.
e: Syntax error: Expecting an expression.
```
I felt like the first 2 messages are essentially the same, but now that I think about it once again,
they are probably not.